### PR TITLE
feat(mango): extract Mango client

### PR DESCRIPTION
**Pull request by rogaldh**
## Description

Removes the Mango v3 instruction decoder and keeps only lightweight **detection**, so the explorer recognises the (dead) Mango v3 protocol and marks it deprecated without bundling `@blockworks-foundation/mango-client` / `@project-serum/serum`. Detection lives in a dedicated `@explorer/decoder-mango` workspace package.

- `packages/decoder-mango` exposes detection only — `isMangoInstruction`, `parseMangoInstructionTitle`, `resolveMangoInstructionName`, `getMangoInstructionLabel`, `MANGO_PROGRAM_IDS`, `MANGO_V3_PROGRAM_LABEL` (plus a `/detection` subpath). The per-instruction decoders, market lookups, hooks, and their fixtures/tests are gone. With the decoder deleted, the heavy `@blockworks-foundation/mango-client` is no longer used anywhere and drops out of the lockfile entirely. (`@project-serum/serum` stays — it's used by the separate Serum decoder, not Mango.)
- The per-instruction Mango cards and the `instruction-program-mango` feature slice are removed. Mango v3 instructions now render through a generic `CommonInstructionDetailsCard` (resolves the program label from the registry, shows raw data), titled e.g. `Mango v3 (deprecated): PlacePerpOrder` — or `No data` / `Unknown` when the payload can't be named.
- `app/utils/programs.ts` registers the three Mango v3 program IDs (devnet / testnet / mainnet) with the deprecated label sourced from the package's `MANGO_V3_PROGRAM_LABEL` — the package is the single source of truth for the addresses + label.
- Transaction history names Mango v3 instructions from the discriminator via the existing `NAME_SOURCES` chain (`resolveMangoInstructionName`) — no payload decoding.

Mango v4 is intentionally out of scope: it's a separate, active Anchor program that resolves its own name/instructions via its on-chain IDL (shown with the existing self-reported indicator), so it needs no bundled decoder or registry entry.

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Protocol integration
-   [ ] Documentation update
-   [x] Other (please describe):

Refactor + deprecation. Replaces the heavy Mango v3 decoder with detection-only recognition and a deprecation notice. Behaviour change: Mango v3 instructions render a generic deprecated card instead of per-instruction detail cards.

## Screenshots

Mango v3 instructions render a single generic card — `Mango v3 (deprecated): <name>` over the raw instruction data — instead of the previous per-instruction detail cards. No new screens; not a protocol integration.

image
image

## Testing

- `packages/decoder-mango` unit tests cover detection across all instruction variants and edge cases (non-mango program, unknown discriminator, empty/short data) for `isMangoInstruction`, `parseMangoInstructionTitle`, `resolveMangoInstructionName`, and `getMangoInstructionLabel`.
- Added Storybook stories for `CommonInstructionDetailsCard` (named / no-data / unknown), covered by the Vitest story sweep.
- `pnpm lint`, `pnpm test`, and app `tsc --noEmit` pass locally; the package `dist` was rebuilt detection-only (no `mango-client`).

## Related Issues

[HOO-452](https://linear.app/solana-fndn/issue/HOO-452)

## Checklist

-   [x] My code follows the project's style guidelines
-   [x] I have added tests that prove my fix/feature works
-   [x] All checks pass locally (`pnpm test`, `pnpm lint`, `pnpm typecheck`)
-   [x] I have run `build:info` script to update build information
-   [x] CI/CD checks pass on the PR
-   [x] Screenshots included — required for protocol screens, recommended for other UI changes

## Additional Notes

- Branch rebased onto latest `master`; `pnpm-lock.yaml` regenerated — the workspace now includes `packages/decoder-mango`, and `@blockworks-foundation/mango-client` is gone entirely (`@project-serum/serum` remains, used by the Serum decoder).
- Mango v4 is deliberately not added to the trusted registry — its name stays self-reported via its IDL, with the existing self-reported indicator.

### CODE REVIEW

File: packages/decoder-mango/src/market.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+const accountInfoCache: Record<string, Promise<AccountInfo<Buffer> | null>> = {};
+function getAccountInfo(clusterUrl: string, publicKey: PublicKey): Promise<AccountInfo<Buffer> | null> {
+    if (publicKey.toBase58() in accountInfoCache) {
+        return accountInfoCache[publicKey.toBase58()];
+    }
+    const connection = new Connection(clusterUrl);
+    const accountInfoPromise = connection.getAccountInfo(publicKey);
+    accountInfoCache[publicKey.toBase58()] = accountInfoPromise;
+    return accountInfoPromise;
+}
```
[the remark this answers is not in the document]
Comment (rogaldh) [reply packages/decoder-mango/src/market.ts]: Addressed

File: packages/decoder-mango/src/market.ts [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        return perpMarketConfigs[0];
+    }
+}
+
+export async function getPerpMarketFromPerpMarketConfig(
+    clusterUrl: string,
+    mangoPerpMarketConfig: PerpMarketConfig,
+): Promise<PerpMarket> {
+    const acc = await getAccountInfo(clusterUrl, mangoPerpMarketConfig.publicKey);
+    const decoded = PerpMarketLayout.decode(acc?.data);
```
[the remark this answers is not in the document]
Comment (rogaldh) [reply packages/decoder-mango/src/market.ts]: Fixed

**Comment by rogaldh** [comment]
@greptile-apps issues were addressed

File: packages/decoder-mango/src/market.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+function getAccountInfo(clusterUrl: string, publicKey: PublicKey): Promise<AccountInfo<Buffer> | null> {
+    const cacheKey = `${clusterUrl}:${publicKey.toBase58()}`;
+    if (cacheKey in accountInfoCache) {
+        return accountInfoCache[cacheKey];
+    }
+    const connection = new Connection(clusterUrl);
+    const accountInfoPromise = connection.getAccountInfo(publicKey);
+    accountInfoCache[cacheKey] = accountInfoPromise;
+    return accountInfoPromise;
+}
```
[the remark this answers is not in the document]
Comment (rogaldh) [reply packages/decoder-mango/src/market.ts]: Fixed

File: app/components/account/TokenHistoryCard.tsx [hunk 4]
```diff
@@ -5,13 +5,13 @@ import { ErrorCard } from '@components/common/ErrorCard';
...
 import { Signature } from '@components/common/Signature';
 import { Slot } from '@components/common/Slot';
-import { isMangoInstruction, parseMangoInstructionTitle } from '@components/instruction/mango/types';
 import { isSerumInstruction, parseSerumInstructionTitle } from '@components/instruction/serum/types';
 import {
     isTokenLendingInstruction,
     parseTokenLendingInstructionTitle,
 } from '@components/instruction/token-lending/types';
 import { isTokenSwapInstruction, parseTokenSwapInstructionTitle } from '@components/instruction/token-swap/types';
+import { isMangoInstruction, parseMangoInstructionTitle } from '@explorer/decoder-mango/detection';
```
Comment (Woody4618) [inline app/components/account/TokenHistoryCard.tsx:17]: Why is this still here?

**Review by askov — APPROVED** [review]

**Review by Woody4618 — APPROVED** [review]
